src/libraries/System.IO.Packaging/src/System/IO/Packaging/OrderedDictionary.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.IO.Packaging
+{
+    /// <summary>
+    /// A collection that ensures uniqueness among a list of elements while maintaining the order in which the elements were added.
+    /// </summary>
+    internal class OrderedDictionary<TKey, TValue> : IEnumerable<TValue>
```

Comment:
We have a `SortedDictionary` class, which is generic and implements `IEnumerable<T>`: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.sorteddictionary-2
Can you use that instead, or was there a reason why you decided against it?

Note: There is an `OrderedDictionary` class, but it's not generic: https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.ordereddictionary

Reply:
Yes, sorted dictionary does not ensure the order is the same as was added - its order is defined by some sort of comparer. The ordering is part of the current behavior and affects the order the relationships are persisted to the file/stream.

Reply:
As to using `System.Collections.Specialized.OrderedDictionary`, there are two issues:

1. It's non-generic. Annoying, but potentially ok. 
2. In order to remove an item, it is searching through the whole array. The implementation I've added just removes a node in the linked list, so it's `O(1)`